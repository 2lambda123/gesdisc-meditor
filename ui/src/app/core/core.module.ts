import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
