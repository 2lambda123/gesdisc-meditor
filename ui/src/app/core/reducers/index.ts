import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
