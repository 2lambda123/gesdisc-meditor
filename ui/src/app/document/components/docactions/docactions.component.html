<ng-container *ngFor="let action of actions">
  <button mat-raised-button (click)="updateState(action.target)">{{ action.label }}</button>
</ng-container>
