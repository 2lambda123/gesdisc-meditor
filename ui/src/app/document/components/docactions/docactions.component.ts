import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Edge } from 'app/service';

@Component({
	selector: 'med-document-actions',
	templateUrl: './docactions.component.html',
	styleUrls: ['./docactions.component.css']
})
export class DocactionsComponent implements OnInit {

	@Input() actions: Edge[];
	@Output() updateState = new EventEmitter<string>();

	constructor() { }

	ngOnInit() {
	}

}
