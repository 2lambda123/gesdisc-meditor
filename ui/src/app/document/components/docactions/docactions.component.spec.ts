import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { DocactionsComponent } from './docactions.component';

describe('DocactionsComponent', () => {
  let component: DocactionsComponent;
  let fixture: ComponentFixture<DocactionsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ DocactionsComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DocactionsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
