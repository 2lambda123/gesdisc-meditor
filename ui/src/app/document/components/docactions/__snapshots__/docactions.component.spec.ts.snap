// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DochistoryComponent should render document actions with mock data 1`] = `
<med-document-actions
  actions={[Function Array]}
  canEdit="false"
  saveDocument={[Function EventEmitter]}
  updateState={[Function EventEmitter]}
>
  <button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 1
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 2
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 3
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</med-document-actions>
`;

exports[`DochistoryComponent should render document actions with mock data 2`] = `
<med-document-actions
  actions={[Function Array]}
  canEdit={[Function Boolean]}
  canSave="false"
  saveDocument={[Function EventEmitter]}
  updateState={[Function EventEmitter]}
>
  <button
    class="mat-raised-button _mat-animation-noopable"
    disabled=""
    mat-raised-button=""
    ng-reflect-disabled="true"
  >
    <span
      class="mat-button-wrapper"
    >
      Save
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="true"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 1
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 2
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 3
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</med-document-actions>
`;

exports[`DochistoryComponent should render document actions with mock data 3`] = `
<med-document-actions
  actions={[Function Array]}
  canEdit={[Function Boolean]}
  canSave={[Function Boolean]}
  saveDocument={[Function EventEmitter]}
  updateState={[Function EventEmitter]}
>
  <button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
    ng-reflect-disabled="false"
  >
    <span
      class="mat-button-wrapper"
    >
      Save
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 1
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 2
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><button
    class="mat-raised-button _mat-animation-noopable"
    mat-raised-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Test 3
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</med-document-actions>
`;
